qiskit/quantum_info/operators/symplectic/pauli.py

```diff
@@ -0,0 +1,1006 @@
+
+    **Matrix Operator Representation**
+
+    Pauli's can be converted to :math:`(2^n, 2^n)`
+    :class:`~qiskit.quantum_info.Operator` using the :meth:`to_operator` method,
+    or to a dense or sparse complex matrix using the :meth:`to_matrix` method.
+
+    **Data Access**
+
+    The individual qubit Paulis can be accessed and updated using the ``[]``
+    operator which accepts integer, lists, or slices for selecting subsets
+    of Paulis. Note that selecting subsets of Pauli's will discard the
+    phase of the current Pauli.
+
+    For example
+
+    .. code:
+
+        p = Pauli('-iXYZ')
+
+        print('P[0] =', repr(P[0]))
+        print('P[1] =', repr(P[1]))
+        print('P[2] =', repr(P[2]))
+        print('P[:] =', repr(P[:]))
+        print('P[::-1] =, repr(P[::-1]))
+    """
+
+    # pylint: disable = missing-param-doc, missing-type-doc
+    def __init__(self, z=None, x=None, phase=None, *, label=None):
+        """Initialize the Pauli.
+
+        When using the symplectic array input data both z and x arguments must
+        be provided, however the first (z) argument can be used alone for string
+        label, Pauli operator, or ScalarOp input data.
+
+        Args:
+            z (array or str or ScalarOp or Pauli): input data or symplectic z vector.
+            x (array): Optional, symplectic x vector.
+            phase (int or None): Optional, phase exponent from Z_4.
+            label (str): DEPRECATED, string label.
+
+        Raises:
+            QiskitError: if input array is invalid shape.
+        """
+        if isinstance(z, BasePauli):
+            # BasePauli initialization
+            base_z, base_x, base_phase = z._z, z._x, z._phase
+            if base_z.shape[0] != 1:
+                raise QiskitError("Input is not a single Pauli")
+        elif x is not None:
+            base_z, base_x, base_phase = self._from_array(z, x, phase)
+            if base_z.shape[0] != 1:
+                raise QiskitError("Input is not a single Pauli")
+        elif isinstance(z, str):
+            base_z, base_x, base_phase = self._from_label(z)
+        elif label is not None:  # DEPRECATED
+            base_z, base_x, base_phase = self._from_label_deprecated(label)
+        elif isinstance(z, ScalarOp):
+            base_z, base_x, base_phase = self._from_scalar_op(z)
+        elif isinstance(z, (QuantumCircuit, Instruction)):
+            base_z, base_x, base_phase = self._from_circuit(z)
+        else:
+            raise QiskitError("Invalid input data for Pauli.")
+
+        # Add phase
+        if phase is not None:
+            base_phase = np.mod(
+                np.sum(np.logical_and(base_z, base_x),
+                       axis=1, dtype=np.int) + phase, 4)
+
+        # Initialize BasePauli
+        super().__init__(base_z, base_x, base_phase)
+
+    def __repr__(self):
+        """Display representation."""
+        return "Pauli('{}')".format(self.__str__())
```

Comment:
While `__str__` is truncated,  `__repr__` should not be truncated.
```suggestion
        return f"Pauli('{self.to_label()}')"
```

Reply:
I was looking this up and python convention is to make `repr` truncated (eg look at repr of a large list, dict or numpy array and they will be truncated). This is import for jupyter notebooks too otherwise you can crash your browser if it trys to display a very large Pauli from a function return or some such

Reply:
I thought repr should be evaluatable, but considering jupyter notebook, you are right. Thank you.

Reply:
I don't think thats a hard rule, but it is nice to have. So repr here is at least evalable when there is no truncation. I also added a class method to specify (or disable) the qubit cuttoff for the string truncation.